src/textual/css/_style_properties.py

```diff
@@ -279,20 +311,22 @@ def __set__(
             StyleValueError: When the supplied ``tuple`` is not of valid length (1, 2, or 4).
         """
         top, right, bottom, left = self._properties
-        obj.refresh()
         if border is None:
             clear_rule = obj.clear_rule
             clear_rule(top)
             clear_rule(right)
             clear_rule(bottom)
             clear_rule(left)
+            obj.refresh(layout=self._layout)
```

Comment:
I suspect that clearing the rule, or using `setattr` will ultimately call `obj.refresh`, so this may be redundant. Can you check?

Reply:
Seems to work without. There was a `refresh` call here before without layout (above this if block - see the diff), I didn't understand what that was for? Do you know?

Reply:
Could be a remnant of a previous system. Managing refreshes is tricky.